### PR TITLE
refactor: simplify module creation

### PR DESCRIPTION
**Pull request by ST-DDT**
Simplifies the modules by reusing the constructors from a shared module base.

This is a v8.x only feature, that will likely be obsolete in v9.0.

What do you think?

### CODE REVIEW

File: src/modules/module-base.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import type { Faker } from '../faker';
+import { bindThisToMemberFunctions } from '../internal/bind-this-to-member-functions';
+import type { SimpleFaker } from '../simple-faker';
+
+/**
+ * Base class for all modules that use a `SimpleFaker` instance.
+ *
+ * @internal
+ */
+export abstract class SimpleModuleBase {
```
Comment (xDivisionByZerox) [inline src/modules/module-base.ts]: Should this file not be located in `src/internal/` if it is marked as internal?

**Review by xDivisionByZerox — CHANGES_REQUESTED** [review]
Can we add extend the test factory (or write a separate one) to include automatic test for these `this` bindings on all methods? Currently I'm not 100% sure that I can do. 

```ts
const method = faker[moduleKey][methodKey]; 
method();
```

Additionally, while being a nice little addition I have a hard time seeing the advantage of introducing it now.

**Comment by ST-DDT** [comment]
> Can we add extend the test factory (or write a separate one) to include automatic test for these `this` bindings on all methods? Currently I'm not 100% sure that I can do.
> 
> ```ts
> const method = faker[moduleKey][methodKey]; 
> method();
> ```

We already have that:

https://github.com/faker-js/faker/blob/a1936938574f574ecc137d9f718a48a2256e49c9/test/support/seededRuns.ts#L146-L147

File: src/modules/module-base.ts (hunk 1 above)
Comment (ST-DDT) [reply src/modules/module-base.ts]: Done

**Review by xDivisionByZerox — DISMISSED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]

**Pull request merged**